azkaban-common/src/main/java/azkaban/project/AzkabanProjectLoader.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.project;
+
+import static java.util.Objects.requireNonNull;
+
+import azkaban.flow.Flow;
+import azkaban.project.ProjectLogEvent.EventType;
+import azkaban.project.validator.ValidationReport;
+import azkaban.project.validator.ValidationStatus;
+import azkaban.project.validator.ValidatorConfigs;
+import azkaban.project.validator.ValidatorManager;
+import azkaban.project.validator.XmlValidatorManager;
+import azkaban.storage.StorageManager;
+import azkaban.user.User;
+import azkaban.utils.Props;
+import azkaban.utils.Utils;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.FileUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class handles the downloading and uploading of projects.
+ */
+class AzkabanProjectLoader {
+
+  private static final Logger log = LoggerFactory.getLogger(AzkabanProjectLoader.class);
+
+  @VisibleForTesting
+  final File tempDir;
+
+  private final Props props;
+  private final ProjectLoader projectLoader;
+  private final StorageManager storageManager;
+  private final int projectVersionRetention;
+
+  @Inject
+  AzkabanProjectLoader(final Props props, final ProjectLoader projectLoader,
+      final StorageManager storageManager) {
+    this.props = requireNonNull(props, "Props is null");
+    this.projectLoader = requireNonNull(projectLoader, "project Loader is null");
+    this.storageManager = requireNonNull(storageManager, "Storage Manager is null");
+
+    tempDir = new File(props.getString("project.temp.dir", "temp"));
+    if (!tempDir.exists()) {
+      log.info("Using temp dir: " + tempDir.getAbsolutePath());
+      tempDir.mkdirs();
+    } else {
+      log.info("Creating temp dir: " + tempDir.getAbsolutePath());
+    }
+    projectVersionRetention = props.getInt("project.version.retention", 3);
```

Comment:
Why isn't this variable prefixed with this.? 

Is it intentional to keep the file as close as the original in this step?